### PR TITLE
refactor: Crm reports cleanup

### PR DESCRIPTION
**Pull request by anupamvs**
**Lead Details**

image

**Lost Opportunity**

image

### CODE REVIEW

File: erpnext/crm/report/lead_details/lead_details.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+			"fieldname": "phone",
+			"fieldtype": "Data",
+			"width": 120
+		},
+		{
+			"label": _("Notes"),
+			"fieldname": "notes",
+			"fieldtype": "Data",
+			"width": 120
+		},
```
Comment (marination) [inline erpnext/crm/report/lead_details/lead_details.py]: Not sure if adding Notes is useful. A report should have compact and to-the-point data. If its long form data, report is not the best way to show that

File: erpnext/crm/report/lead_details/lead_details.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+def get_conditions(filters) :
+	conditions = []
+
+	if filters.get("territory"):
+		conditions.append("territory=%(territory)s")
+
+	if filters.get("status"):
+		conditions.append("status=%(status)s")
+
+	return " and {}".format(" and ".join(conditions)) if conditions else ""
```
Comment (marination) [inline erpnext/crm/report/lead_details/lead_details.py]:
```suggestion
		conditions.append("and territory=%(territory)s")

	if filters.get("status"):
		conditions.append("and status=%(status)s")

	return conditions if conditions else ""
```

File: erpnext/crm/report/lost_opportunity/lost_opportunity.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+def get_data(filters):
+	return frappe.db.sql("""
+		SELECT
+			`tabOpportunity`.name,
+			`tabOpportunity`.opportunity_from,
+			`tabOpportunity`.party_name,
+			`tabOpportunity`.customer_name,
+			`tabOpportunity`.opportunity_type,
+			`tabOpportunity`.contact_by
```
Comment (marination) [inline erpnext/crm/report/lost_opportunity/lost_opportunity.py]: You can also fetch:
-  Lost Reason to analyse trends in why opportunities are lost
- Territory, to know where the most loss is happening
- Sales Stage to know where in the sales cycle improvement can happen
Just a suggestion

File: erpnext/crm/report/lost_opportunity/lost_opportunity.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	if filters.get("opportunity_from"):
+		conditions.append("opportunity_from=%(opportunity_from)s")
+
+	if filters.get("party_name"):
+		conditions.append("party_name=%(party_name)s")
+
+	if filters.get("contact_by"):
+		conditions.append("contact_by=%(contact_by)s")
+
+	return " and {}".format(" and ".join(conditions)) if conditions else ""
```
Comment (marination) [inline erpnext/crm/report/lost_opportunity/lost_opportunity.py]: here also, you prefix 'and' before each condition as parent query has conditions already. And `return conditions if conditions else "" `. Simpler the better

**Review by marination — CHANGES_REQUESTED** [review]

File: erpnext/crm/report/lost_opportunity/lost_opportunity.py (hunk 3 above)
Comment (anupamvs) [reply erpnext/crm/report/lost_opportunity/lost_opportunity.py]: @marination Done!

**Review by marination — DISMISSED** [review]

**Comment by anupamvs** [comment]
@marination docs https://github.com/frappe/erpnext_documentation/pull/133

**Review by marination — APPROVED** [review]

**Pull request merged**